### PR TITLE
feat: i18n for home

### PR DESCRIPTION
**Pull request by im-adithya**
### Describe the changes you have made in this PR

i18n for home screen [and the components present in Home Screen]

### Link this PR to an issue

#1283

### Type of change (Remove other not matching type)

- `feat`: New feature (non-breaking change which adds functionality)

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

File: src/app/components/AllowanceMenu/index.tsx [hunk 1]
```diff
@@ -97,13 +100,13 @@ function AllowanceMenu({ allowance, onEdit, onDelete }: Props) {
...
         onRequestClose={closeModal}
-        contentLabel="Allowance Options"
+        contentLabel={t("options_label")}
         overlayClassName="bg-black bg-opacity-25 fixed inset-0 flex justify-center items-center p-5"
         className="rounded-lg bg-white w-full max-w-lg"
       >
         <div className="p-5 flex justify-between dark:bg-surface-02dp">
           <h2 className="text-2xl font-bold dark:text-white">
-            Set a new budget
+            {t("set_budget")}
```
Comment (escapedcat) [inline src/app/components/AllowanceMenu/index.tsx:109]: Can this be `title`? Or title of a subsection?

File: src/app/components/AllowanceMenu/index.tsx [hunk 2]
```diff
@@ -120,11 +123,11 @@ function AllowanceMenu({ allowance, onEdit, onDelete }: Props) {
             <div className="w-60">
               <DualCurrencyField
                 id="budget"
-                label="New budget"
+                label={t("new_budget_label")}
```
Comment (escapedcat) [inline src/app/components/AllowanceMenu/index.tsx:126]: Could this be `new_budget.label` and below `new_budget.hint`?

File: src/app/components/AllowanceMenu/index.tsx [hunk 3]
```diff
@@ -97,13 +100,13 @@ function AllowanceMenu({ allowance, onEdit, onDelete }: Props) {
         closeTimeoutMS={200}
         isOpen={modalIsOpen}
         onRequestClose={closeModal}
-        contentLabel="Allowance Options"
+        contentLabel={t("options_label")}
```
Comment (escapedcat) [inline src/app/components/AllowanceMenu/index.tsx:103]: I always feel using `_` as a separator of "name" and "type" feels off. Feels cleaner to do `options.label`. In this case it's easier to extend it in the future if needed. What do you think?

File: src/app/components/AllowanceMenu/index.tsx (hunk 1 above)
Comment (im-adithya) [reply src/app/components/AllowanceMenu/index.tsx:109]: Title of subsection ig as this is the Allowance Menu component.

File: src/app/components/AllowanceMenu/index.tsx (hunk 2 above)
Comment (im-adithya) [reply src/app/components/AllowanceMenu/index.tsx:126]: I was actually about to use them but then thought all other components don't extend further as in nested objects in translations so I thought of keeping it simple without nesting...

Comment (escapedcat) [reply src/app/components/AllowanceMenu/index.tsx:126]: :D Let's nest it

**Review by escapedcat — APPROVED** [review]

**Pull request merged**